ten-simple-rules-dockerfiles.Rmd

```diff
@@ -119,10 +119,14 @@ Container images include the application, e.g., the programming language interpr
 A `Dockerfile` consists of a sequence of instructions to copy files and install software.
 Each instruction adds a layer to the image, which can be cached across image builds for minimizing build and download times.
 The images have a main executable exposed as an "entrypoint" that is started when they are run as stateful containers, which are the running instances of Docker images.
-Containers can be modified, stopped, restarted and purged.
+Containers can be modified, stopped, restarted and purged. A visual analogy with a compiled programming language is given in Figure&nbsp;1 for the workflow used to specify and build running Docker containers. 
 `Dockerfile`s, like Makefiles [@wikipedia_contributors_make_2019], are meant to be useful for both humans to read and for computers to execute.
 See Listing&nbsp;1 for a full `Dockerfile`, which we will refer to throughout this article.
 
+```{r container-analogy, echo=FALSE, fig.cap="The workflow to create Docker containers by analogy. Containers begin with a Dockerfile, a recipe for building the computational environment (analogous to source code in a compiled programming language). This is used to build an image with the `docker build` command (analogous to compiling the source code into a binary). Finally the image is used to launch one or more containers with the `docker run` command (analogous to running an instance of the compiled binary as a process).", out.width = '100%'}
+knitr::include_graphics("container-analogy.png")
```

Comment:
I spoke too soon! I didn't see this description. Disregard the previous comment if you think it's sufficient. I might still add a little description since the previous text is in the paper and this is a figure caption. Up to you!

Reply:
I've added some more description in the main text.